### ISSUE TITLE
[Feature request] Logging Level and Progress Bar for Model Downloads

### ISSUE THREAD

**Issue by aress31** [xenova/transformers.js#117]
Right now, the output can be quite lengthy and verbose, see:

![image](https://github.com/xenova/transformers.js/assets/11601622/3fa55cb7-7cd0-4b2b-80cd-3fcea20ace55)

Could it be possible to expose logger options to granularly control the output as well as offering a visual feedback for the status of the model download from HuggingFace?

**Comment by xenova** [comment]
I agree - that would be a great improvement! It was not originally considered since the library was designed for browsers, but since there has been a lot of interest for Node.js-like environments, it's definitely something to consider.

**Comment by aress31** [comment]
That is exactly my use case, that would be for a `Node` application. And as such, I need a way to provide some visual feedback on the different operations, something similar in essence to:
- https://huggingface.co/docs/transformers/main_classes/logging

**Comment by xenova** [comment]
A global `env.logLevel` should be sufficient, right?

**Comment by kungfooman** [comment]
I kinda lend to extending the options in `pipeline()`, because I would rather minimize the amount of global env state/variables :thinking: 

https://github.com/microsoft/onnxruntime/blob/18f17c555d51caee83f15983c2620f463fbaddd1/js/common/lib/inference-session.ts#L126-L139

We could just make it a passthrough-options-ONNX-session-options-object, in the case that we need to use other session options later aswell, inside the existing pipeline options object.

Something like:

```js
/**
 * Utility factory method to build a [`Pipeline`] object.
 *
 * @param {string} task The task of the pipeline.
 * @param {string} [model=null] The name of the pre-trained model to use. If not specified, the default model for the task will be used.
 * @param {PretrainedOptions} [options] Optional parameters for the pipeline.
 * @returns {Promise<Pipeline>} A Pipeline object for the specified task.
 * @throws {Error} If an unsupported pipeline is requested.
 */
export async function pipeline(
    task,
    model = null,
    {
        quantized = true,
        progress_callback = null,
        config = null,
        cache_dir = null,
        local_files_only = false,
        revision = 'main',
        ortExtraSessionOptions = {},
    } = {}
) {
```


But we could make the API easier to use aswell (e.g. using `"verbose"` instead of `0` because no one wants to memorize enums):

Do we even need both options, `logVerbosityLevel` and `logSeverityLevel`? It's not clear to me yet what the difference is :see_no_evil: 

I have a slight feeling that other dev's are also a bit confused:

![image](https://github.com/xenova/transformers.js/assets/5236548/8218b51f-6452-4495-96fb-1b284e294e63)

**Comment by xenova** [comment]
> Do we even need both options, logVerbosityLevel and logSeverityLevel? It's not clear to me yet what the difference is 🙈

My thought exactly. This is why a global logging level might be okay. It will also be better for when we add more backends (not just onnx). If someone REALLY wants those log levels, they can set them manually with `env.backends.onnx.*`

**Comment by AmitDJagtap** [comment]
For people looking for the final answer / code.  adding a nodejs code snippet here to set log level to 3 . 

warning levels
  ```
  VERBOSE = 0,
  INFO = 1,
  WARNING = 2,
  ERROR = 3,
  FATAL = 4
```
your code 
```

import { pipeline, env } from '@xenova/transformers';
import fs from 'fs';
env.cacheDir = './.cache';
env.backends.onnx.logSeverityLevel = 3; // this line here
```

**Comment by fcuenya** [comment]
Is there anything else to work on this feature? I would like to contribute but it is not clear to me if this needs further work. Thank you!
